modules/lang-groovy/src/test/resources/rest-api-spec/test/lang_groovy/30_compile_limit.yaml

```diff
@@ -0,0 +1,53 @@
+setup:
+  - do:
+      indices.create:
+          index:  test
+  - do:
+      index:
+          index:  test
+          type:   test
+          id:     1
+          body:   { foo: bar }
+          refresh: wait_for
+
+---
+"circuit breaking with too many scripts":
+
+  - do:
+      cluster.put_settings:
+        body:
+          transient:
+            script.max_compilations_per_minute: 1
+
+  - do:
+      search:
+        index: test
+        type:  test
+        body:
+          size: 1
+          script_fields:
+            myfield:
+              script:
+                inline: "\"aoeu\""
+                lang: groovy
+
+  - match: { hits.total: 1 }
+
+  - do:
+      catch: /Too many dynamic script compilations within one minute/
+      search:
+        index: test
+        type:  test
+        body:
+          size: 1
+          script_fields:
+            myfield:
+              script:
+                inline: "\"aoeuaoeu\""
+                lang: groovy
+
+  - do:
```

Comment:
I wonder if you can use `teardown` for this that way it'll get unset after the test.